pkg/ec2metadata/ec2metadata.go

```diff
@@ -325,30 +325,43 @@ func retry(attempts int, sleep time.Duration, httpReq func() (*http.Response, er
 }
 
 // GetNodeMetadata attempts to gather additional ec2 instance information from the metadata service
-func (e *Service) GetNodeMetadata() NodeMetadata {
+func (e *Service) GetNodeMetadata(imdsDisabled bool) NodeMetadata {
 	var metadata NodeMetadata
-	identityDoc, err := e.GetMetadataInfo(IdentityDocPath)
-	if err != nil {
-		log.Err(err).Msg("Unable to fetch metadata from IMDS")
-		return metadata
-	}
-	err = json.NewDecoder(strings.NewReader(identityDoc)).Decode(&metadata)
-	if err != nil {
-		log.Warn().Msg("Unable to fetch instance identity document from ec2 metadata")
-		metadata.InstanceID, _ = e.GetMetadataInfo(InstanceIDPath)
-		metadata.InstanceType, _ = e.GetMetadataInfo(InstanceTypePath)
-		metadata.LocalIP, _ = e.GetMetadataInfo(LocalIPPath)
-		metadata.AvailabilityZone, _ = e.GetMetadataInfo(AZPlacementPath)
+        if (!imdsDisabled) {
+            identityDoc, err := e.GetMetadataInfo(IdentityDocPath)
+            if err != nil {
+	        log.Err(err).Msg("Unable to fetch metadata from IMDS")
+	        return metadata
+            }
+            err = json.NewDecoder(strings.NewReader(identityDoc)).Decode(&metadata)
+            if err != nil {
+                log.Warn().Msg("Unable to fetch instance identity document from ec2 metadata")
+                metadata.InstanceID, _ = e.GetMetadataInfo(InstanceIDPath)
+                metadata.InstanceType, _ = e.GetMetadataInfo(InstanceTypePath)
+	        metadata.LocalIP, _ = e.GetMetadataInfo(LocalIPPath)
+	        metadata.AvailabilityZone, _ = e.GetMetadataInfo(AZPlacementPath)
 		if len(metadata.AvailabilityZone) > 1 {
-			metadata.Region = metadata.AvailabilityZone[0 : len(metadata.AvailabilityZone)-1]
+		        metadata.Region = metadata.AvailabilityZone[0 : len(metadata.AvailabilityZone)-1]
 		}
-	}
-	metadata.InstanceLifeCycle, _ = e.GetMetadataInfo(InstanceLifeCycle)
-	metadata.LocalHostname, _ = e.GetMetadataInfo(LocalHostnamePath)
-	metadata.PublicHostname, _ = e.GetMetadataInfo(PublicHostnamePath)
-	metadata.PublicIP, _ = e.GetMetadataInfo(PublicIPPath)
+	    }
+            metadata.InstanceLifeCycle, _ = e.GetMetadataInfo(InstanceLifeCycle)
+	    metadata.LocalHostname, _ = e.GetMetadataInfo(LocalHostnamePath)
+	    metadata.PublicHostname, _ = e.GetMetadataInfo(PublicHostnamePath)
+	    metadata.PublicIP, _ = e.GetMetadataInfo(PublicIPPath)
 
-	log.Info().Interface("metadata", metadata).Msg("Startup Metadata Retrieved")
+	    log.Info().Interface("metadata", metadata).Msg("Startup Metadata Retrieved")
+        } else {
+            metadata.AccountId         = ""
+	    metadata.InstanceID        = ""
+	    metadata.InstanceLifeCycle = ""
+	    metadata.InstanceType      = ""
+	    metadata.PublicHostname    = ""
+	    metadata.PublicIP          = ""
+	    metadata.LocalHostname     = ""
+	    metadata.LocalIP           = ""
+	    metadata.AvailabilityZone  = ""
+	    metadata.Region            = ""
+        }
```

Comment:
Is this `else` branch necessary? When `imdsDisabled=false`, `metadata` should still be empty, since no properties were set. When you were testing, did you discover this was required?

Reply:
He complained about metadata being uninitialized so I've set it to at least empty string values.

Reply:
Who is "he"?

Reply:
The aws-node-termination-handler process. I ran it both on AWS EKS and on my workstation's local k8s cluster.